noxfile.py

```diff
@@ -59,3 +64,151 @@ def docs(session):
             "docs",  # source directory
             "docs/_build/" + dest,  # output directory
         )
+
+
+@nox.session
+def release(session):
+    package_name = "packaging"
+    version_file = Path(f"{package_name}/__about__.py")
+
+    try:
+        release_version = _get_version_from_arguments(session.posargs)
+    except ValueError as e:
+        session.error(f"Invalid arguments: {e}")
+
+    # Check state of working directory and git
+    _check_working_directory_state(session)
+    _check_git_state(session, release_version)
+
+    # Bump to release version
+    _bump(session, version=release_version, file=version_file, kind="release")
+
+    # Tag the release commit
+    session.run("git", "tag", "-s", release_version, external=True)
+
+    # Bump for development
+    major, minor = map(int, release_version.split("."))
+    next_version = f"{major}.{minor + 1}.dev0"
+    _bump(session, version=next_version, file=version_file, kind="development")
+
+    # Checkout the git tag
+    session.run("git", "checkout", "-q", release_version, external=True)
+
+    # Build the distribution
+    session.run("python", "setup.py", "sdist", "bdist_wheel")
+
+    # Check what files are in dist/ for upload.
+    files = glob.glob(f"dist/*")
+    assert sorted(files) == [
+        f"dist/{package_name}-{release_version}.tag.gz",
+        f"dist/{package_name}-{release_version}-py2.py3-none-any.whl",
+    ], f"Got the wrong files: {files}"
+
+    # Get back out into master
+    session.run("git", "checkout", "-q", "master", external=True)
+
+    # Check and upload distribution files
+    session.run("twine", "check", *files)
+
+    # Upload the distribution
+    session.run("twine", "upload", *files)
+
+    # Push the commits and tag
+    # NOTE: The following fails if pushing to the branch is not allowed. This can
+    #       happen on GitHub, if the master branch is protected, there are required
+    #       CI checks and "Include administrators" is enabled on the protection.
+    session.run("git", "push", "upstream", "master", release_version, external=True)
+
+
+# -----------------------------------------------------------------------------
+# Helpers
+# -----------------------------------------------------------------------------
+def _get_version_from_arguments(arguments):
+    """Checks the arguments passed to `nox -s release`.
+
+    Only 1 argument that looks like a version? Return the argument.
+    Otherwise, raise a ValueError describing what's wrong.
+    """
+    if len(arguments) != 1:
+        raise ValueError("Expected exactly 1 argument")
+
+    version = arguments[0]
+    parts = version.split(".")
+
+    if len(parts) != 2:
+        # Not of the form: YY.N
+        raise ValueError("not of the form: YY.N")
+
+    if not all(part.isdigit() for part in parts):
+        # Not all segments are integers.
+        raise ValueError("non-integer segments")
+
+    # All is good.
```

Comment:
Worth checking that the major version is `9 < major < 100` to basically make sure it's double digits?

Reply:
I mean, because we *know* that packaging is CalVer, I guess we could add a constraint of `major == current_year % 100` here.

This can be done in a follow up though, and I'd prefer that. :)